src/Microsoft.AspNet.OData.Shared/Formatter/Deserialization/ODataResourceSetWrapperBase.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Microsoft.OData;
+
+namespace Microsoft.AspNet.OData.Formatter.Deserialization
+{
+    /// <summary>
+    /// Encapsulates an <see cref="ODataResourceSet"/> and the <see cref="ODataResource"/>'s that are part of it.
+    /// </summary>
+    public abstract class ODataResourceSetWrapperBase : ODataItemBase
+    {
+        /// <summary>
+        /// /
+        /// </summary>
+        public ResourceSetType ResourceSetType;
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="ODataResourceSetWrapper"/>.
+        /// </summary>
+        /// <param name="item">The wrapped item.</param>
+        public ODataResourceSetWrapperBase(ODataResourceSetBase item)
+            : base(item)
+        {
+            ResourceItems = new List<ODataItemBase>();            
+        }
+
+   
+         /// <summary>
+        /// Gets the wrapped <see cref="ResourceSetBase"/>.
+        /// </summary>
+        public ODataResourceSetBase ResourceSetBase
+        {
+            get
+            {
+                return Item as ODataResourceSetBase;             
+            }
+        }
+
+        /// <summary>
+        /// Gets the nested resources and deltalinks of this ResourceSet.
+        /// </summary>
+        protected IList<ODataItemBase> ResourceItems { get; private set; }
+
+        /// <summary>
+        /// 
+        /// </summary>
```

Comment:
please fill in.

Reply:
Summary still not filled in in iteration 9.

---
In reply to: [518457966](https://github.com/OData/WebApi/pull/2275#discussion_r518457966) [](ancestors = 518457966)